src/erlfmt.erl

```diff
@@ -19,8 +19,16 @@
     init/1,
     format_file/2,
     format_file_range/4,
+    format_file_range_extract/4,
     format_string/2,
     format_string_range/4,
+    format_string_range_extract/4,
+    % FIXME: We are exporting this function to quiet dialyzer.
```

Comment:
What exactly was the dialyzer error, sorry I didn't have time to check before
We should try to avoid big comments like this and try to fix it.

Reply:
Cf https://github.com/WhatsApp/erlfmt/runs/2678261048
You can also reproduce locally by removing the export!

Reply:
Apparently this function is ok. But when the code path is from `format_file_range` or `format_string_range`, dialyzer infers that passed `Result` cannot be `{ok, Formatted, Info}`, while the unit tests show otherwise.

I have no idea how/why dialyzer infers that. @michalmuskala, any clue?

Reply:
What I usually do is start adding specs down the call stack to try and debug dialyzer error.  They are cryptic and seem unreasonable sometimes, but they have also found a bug for me, at least.